addons/sale_coupon/models/sale_order.py

```diff
@@ -294,7 +294,12 @@ def _get_reward_line_values(self, program):
             return [self._get_reward_values_product(program)]
 
     def _create_reward_line(self, program):
-        self.write({'order_line': [(0, False, value) for value in self._get_reward_line_values(program)]})
+        values = self._get_reward_line_values(program)
+        last_sequence = max(self.order_line.mapped('sequence'))
+        for value in values:
+            last_sequence += 1
+            value['sequence'] = last_sequence
+        self.write({'order_line': [(0, False, value) for value in values]})
```

Comment:
We do not actually need for each line of our reward to have a different sequence.
Should `_get_reward_line_values` be generated with the right sequence in the first place?
```suggestion
        values = self._get_reward_line_values(program)
        seq = max(self.order_line.mapped('sequence'), default=10) + 1
        for value in values:
            value['sequence'] = seq
        self.write({'order_line': [(0, False, value) for value in values]})
```